pkg/snmp/traps/formatter.go

```diff
@@ -340,12 +340,17 @@ func formatType(variable gosnmp.SnmpPDU) string {
 }
 
 func formatValue(variable gosnmp.SnmpPDU) interface{} {
-	switch variable.Value.(type) {
-	case []byte:
-		return string(variable.Value.([]byte))
-	default:
-		return variable.Value
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Is there are particular reasons why we can't keep the `switch` syntax? and do something like this:
```go
func formatValue(variable gosnmp.SnmpPDU) interface{} {
	if variable.Type == gosnmp.ObjectIdentifier {
		return NormalizeOID(value.(string))
	}
	switch variable.Value.(type) {
	case []byte:
		return string(variable.Value.([]byte))
	default:
		return variable.Value
	}
}
```

Reply:
Or maybe something like this:

```
func formatValue(variable gosnmp.SnmpPDU) interface{} {
	switch variable.Value.(type) {
	case []byte:
		return string(variable.Value.([]byte))
    case string:
        if variable.Type == gosnmp.ObjectIdentifier {
		    return NormalizeOID(value.(string))
	    }
        return variable.Value
	default:
		return variable.Value
	}
}
```

Should protect against value not being a string type.

Reply:
Yes we can keep the switch syntax, but shouldn't the `variable.Type == gosnmp.ObjectIdentifier` condition be checked after the switch ?

Reply:
missed your second comment, noted!